sycl/test/devicelib/cmath_fp64_test.cpp

```diff
@@ -0,0 +1,113 @@
+// UNSUPPORTED: windows
+// RUN: %clangxx -fsycl -c %s -o %t.o
+// RUN: %clangxx -fsycl %t.o %sycl_libs_dir/libsycl-cmath-fp64.o -o %t.out
```

Comment:
Please add a RUN line for execution here and in all other tests.
Otherwise this just tests for compilation and no runtime validation is performed.

Reply:
If these are going to run the compiled program, they should be moved to end-to-end testing.

Reply:
Hi, @andykaylor  and @asavonic 
Could we enable the run command for cmath tests on CPU device currently?
For GPU, only OCL RT can work. For CPU, all cmath and complex tests should work in theory but OCL CPU RT has an issue which will lead to crash on some platforms, the fix has not been latest OCL CPU RT, so we may encounter some unexpected crash if enable complex tests.
Thank you very much.

Reply:
@andykaylor 
> If these are going to run the compiled program, they should be moved to end-to-end testing.

Most of the tests in `sycl/test` directory have RUN lines to test on available devices.
This test in particular has a lot of runtime asserts.

> Could we enable the run command for cmath tests on CPU device currently?

I think they should be enabled for all devices you expect this code to work. I assume it should work for CPU and GPU, right? FPGA is probably not going  work, because AOT compilation is not support for device libraries yet.

Reply:
Yes, the tests are expected to work on both CPU and GPU. However, for GPU, we need to force the tests to run on OCL GPU runtime as L0 hasn't supported device library. For CPU, if we enable complex tests, crash may happen on some platform due to OCL CPU runtime issues. So, for GPU, can we add export SYCL_BE=PI_OPENCL?

Reply:
If a test doesn't work on some platform due to a bug, it should be disabled temporarily.